TheWarWithin/ShamanRestoration.lua

```diff
@@ -660,10 +660,8 @@ spec:RegisterAbilities( {
         charges = function()
             if talent.healing_stream_totem.rank + talent.healing_stream_totem_2.rank > 1 then return 2 end
         end,
-        cooldown = 30,
-        recharge = function()
-            if talent.healing_stream_totem.rank + talent.healing_stream_totem_2.rank > 1 then return 30 end
-        end,
+        cooldown = function () return 30 - talent.totemic_surge.enabled and 6 or 0 end,
+        recharge = function() return 30 - talent.totemic_surge.enabled and 6 or 0 end,
```

Comment:
Make `recharge` return nil if you don't have both Healing Stream Totem talents (makes CD calculation less expensive).